File: src/com/jetbrains/idear/ParserService.java
package com.jetbrains.idear;

import com.intellij.openapi.components.ServiceManager;
import opennlp.tools.parser.Parse;
import org.jetbrains.annotations.Nullable;

public abstract class ParserService {

    public void init() {
    }

    public void dispose() {
    }

    @Nullable
    public abstract Parse parseSentence(String sentence);

    public static ParserService getInstance() {
        return ServiceManager.getService(ParserService.class);
    }

}

File: src/main/java/com/jetbrains/idear/nlp/ParserService.kt
package com.jetbrains.idear.nlp

import com.intellij.openapi.components.ServiceManager
import opennlp.tools.parser.Parse

abstract class ParserService {
    open fun init() {}

    abstract fun parseSentence(sentence: String): Parse?

    companion object {
        val instance: ParserService
            get() = ServiceManager.getService(ParserService::class.java)
    }
}
